src/vtk/PolygonWidget/behavior.ts

```diff
@@ -84,23 +106,25 @@ export default function widgetBehavior(publicAPI: any, model: any) {
     return false;
   }
 
-  function updateActiveStateHandle(callData: any) {
+  function getWorldCoords(callData: any) {
     const manipulator =
       model.activeState?.getManipulator?.() ?? model.manipulator;
     if (!manipulator) {
-      return macro.VOID;
+      return undefined;
     }
 
-    const worldCoords = manipulator.handleEvent(
-      callData,
-      model._apiSpecificRenderWindow
-    );
+    return manipulator.handleEvent(callData, model._apiSpecificRenderWindow);
+  }
+
+  function updateActiveStateHandle(callData: any) {
+    const worldCoords = getWorldCoords(callData);
 
     if (
+      worldCoords &&
       worldCoords.length &&
```

Comment:
Nit: this can be simplified to `worldCoords?.length`

Reply:
🙏

Reply:
Found a couple more logic simplifications along `?` lines.